doc/source/examples/metrology/Creating Network Definitions for SOLT Calibration Standards.ipynb

```diff
@@ -0,0 +1,590 @@
+    "medium = Coaxial(frequency=freq, z0=50)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Modeling the Offset Transmission Line\n",
+    "\n",
+    "An offset delay is easily created by the `medium.line()` method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "offset_open  = medium.line(d=29.242, unit='ps')\n",
+    "offset_short = medium.line(d=31.785, unit='ps')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The offset loss is more tricky. We need to model it as a resistor\n",
+    "but it's frequency-dependent and cannot be represented as a single\n",
+    "value. Fortunately, the `medium.resistor()` method conveniently\n",
+    "accepts an array just for this purpose.\n",
+    "\n",
+    "In earlier scikit-rf versions, the method only accepts a `numpy`\n",
+    "array, not a Python list, so a type conversion to `np.array()`\n",
+    "is required."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "resistor_open = []\n",
+    "for f in freq.f:\n",
+    "        r = 2.2 * 29.242 * np.sqrt(f / 1e9)\n",
+    "        resistor_open += [r]\n",
+    "loss_open = medium.resistor(np.array(resistor_open))\n",
+    "\n",
+    "resistor_short = []\n",
+    "for f in freq.f:\n",
+    "        r = 2.36 * 31.785 * np.sqrt(f / 1e9)\n",
+    "        resistor_short += [r]\n",
+    "loss_short = medium.resistor(np.array(resistor_short))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Modeling the Shunt Impedance\n",
+    "\n",
+    "Next, we need to model the shunt impedance of the open and short standard.\n",
+    "For the open standard, it's a capacitance. For the short standard, it's\n",
+    "an inductance.\n",
+    "\n",
+    "Both are modeled as a third-degree polynomial, as a function of frequency.\n",
+    "so let's define a general polynomial function first."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def poly3(freq, x0, x1, x2, x3):\n",
+    "    return x0 + x1 * freq + x2 * freq ** 2 + x3 * freq ** 3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can generate the frequency-dependent capacitors and inductors in\n",
+    "a way similar to how the resistors were generated. It's a shunt\n",
+    "impedance, so one should use `shunt_capacitor()` and `shunt_inductor()` method\n",
+    "(not `capacitor()` and `inductor()`, which are series impedances)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "capacitor_open = []\n",
+    "for f in freq.f:\n",
+    "        capacitor_open += [poly3(f, 49.43, -310.1, 23.17, -0.1597)]\n",
```

Comment:
You should be able to replace all these for loops over frequency by just passing `freq.f` as a parameter to the function. For example: `capacitor_open = poly3(freq.f, 49.43, -310.1, 23.17, -0.1597)`

Reply:
Eventually, NumPy also provides convenient polynomial fit functions:

```python
f = np.array([10, 20, 30])
c = [5, 4, 3, 2]  # polynomial’s coefficients in decreasing powers
p = np.poly1d(c)  
print(p(f)) # same as poly3(f, 2, 3, 4, 5)
```